### PR TITLE
Update documentation about using binary data

### PR DESCRIPTION
**Pull request by Pessimistress**
For #3758 

#### Change List
- Docs and developer guide

### CODE REVIEW

File: docs/api-reference/layer.md [hunk 1]
```diff
@@ -62,7 +62,24 @@ deck.gl layers typically expect `data` to be one of the following types:
 - `Promise`: the resolved value will be used as the value of the `data` prop.
 - `AsyncIterable`: an [async iterable](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/asyncIterator) object that yields data in batches. The default implementation expects each batch to be an array of data objects; one may change this behavior by supplying a custom `dataTransform` callback.
 
-Remarks:
+**data.attributes**
+
+When using a non-iterable `data` object, the object may optionally contain a field `attributes`, if the application wishes to supply binary buffers directly to the layer. This use case is discussed in detail in the [performance developer guide](/docs/developer-guide/performance.md#supply-attributes-directly).
+
+The keys in `data.attributes` correspond to the [accessor](/docs/developer-guide/using-layers.md#accessors) name that the binary should replace, for example `getPosition`, `getColor`. See each layer's documentation for available accessor props.
```
Comment (1chandu) [inline docs/api-reference/layer.md]: Isn't the key `attribute name` instead of `accessor`?

Comment (Pessimistress) [reply docs/api-reference/layer.md]: No. This is the new feature in v8.

File: docs/layers/path-layer.md [hunk 2]
```diff
@@ -175,6 +175,67 @@ The width of each path, in units specified by `widthUnits` (default meters).
 * If a number is provided, it is used as the width for all paths.
 * If a function is provided, it is called on each path to retrieve its width.
 
+
+## Use binary attributes
+
+This section is about the special requirements when [supplying attributes directly](/docs/developer-guide/performance.md#supply-attributes-directly) to a `PathLayer`.
+
+Because each path have different number of vertices, when `data.attributes.getPath` is supplied, the layer also requires an array `data.startIndices` that describes the vertex index at the start of each path. For example, if there are 3 paths of 2, 3, and 4 vertices each, `startIndices` should be `[0, 2, 5, 9]`.
```
Comment (tsherif) [inline docs/layers/path-layer.md]: "each path has a different number of"

File: docs/layers/solid-polygon-layer.md [hunk 3]
```diff
@@ -163,6 +163,69 @@ Only applies if `extruded: true`.
 * If a number is provided, it is used as the elevation for all polygons.
 * If a function is provided, it is called on each object to retrieve its elevation.
 
+
+## Use binary attributes
+
+This section is about the special requirements when [supplying attributes directly](/docs/developer-guide/performance.md#supply-attributes-directly) to a `SolidPolygonLayer`.
+
+Because each polygon have different number of vertices, when `data.attributes.getPolygon` is supplied, the layer also requires an array `data.startIndices` that describes the vertex index at the start of each path. For example, if there are 3 polygons of 3, 4, and 5 vertices each (including the end vertex that overlaps with the first vertex to close the loop), `startIndices` should be `[0, 3, 7, 12]`. *Polygons with holes are not supported when using precalculated attributes.*
```
Comment (tsherif) [inline docs/layers/solid-polygon-layer.md]: "each polygon has a different number of"

**Pull request merged**